include/zephyr/bluetooth/audio/hap.h

```diff
@@ -0,0 +1,394 @@
+
+	/**
+	 * @brief Controller info has been updated.
+	 *
+	 * This callback notifies the application that the controller info has been changed,
+	 * e.g. the Hearing Aid Features has been updated.
+	 *
+	 * @param conn Controller instance pointer.
+	 * @param info Info object.
+	 */
+	void (*info)(struct bt_hap_harc *harc, const struct bt_hap_harc_info *info);
+
+	sys_snode_t _node;
+};
+
+/**
+ * @brief Register Hearing Aid Remote Controller callbacks.
+ *
+ * @param cb The callback structure.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_hap_harc_cb_register(struct bt_hap_harc_cb *cb);
+
+/**
+ * @brief Unregister Hearing Aid Remote Controller callbacks.
+ *
+ * @param cb The callback structure.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_hap_harc_cb_unregister(struct bt_hap_harc_cb *cb);
+
+/**
+ * @brief Bind profile.
+ *
+ * Bind profile to remote device implementing HAP Hearing Aid role.
+ *
+ * @param conn Bluetooth connection object.
+ * @param[out] harc Valid controller object on success.
+ *
+ * @return Zero on success or (negative) error code on failure.
+ */
+int bt_hap_harc_bind(struct bt_conn *conn, struct bt_hap_harc **harc);
+
+/**
+ * @brief Unbind profile.
+ *
+ * Unbind profile or cancel pending profile binding.
+ *
+ * @param harc Controller instance pointer.
+ *
+ * @return Zero on success or (negative) error code on failure.
+ */
+int bt_hap_harc_unbind(struct bt_hap_harc *harc);
+
+/** @brief Hearing Aid Remote Controller Preset List callback structure. */
+struct bt_hap_harc_preset_cb {
+	/**
+	 * @brief Callback function for Hearing Access Service active preset changes.
+	 *
+	 * Optional callback called when the active preset is changed by the remote server when the
+	 * preset switch procedure is complete. The callback must be set to receive active preset
+	 * changes and enable support for switching presets.
+	 *
+	 * @param harc Controller instance pointer.
+	 * @param index Active preset index.
+	 */
+	void (*active)(struct bt_hap_harc *harc, uint8_t index);
+
+	/**
+	 * @brief Callback function for preset store.
+	 *
+	 * The callback is called when the Preset Record Update ot Preset Read Response is notified
+	 * by the remote server.
+	 * The record object as well as its objects are temporary and must be copied to in order
+	 * to cache its information.
+	 *
+	 * @param harc Controller instance pointer.
+	 * @param record Preset record to store.
+	 */
+	void (*store)(struct bt_hap_harc *harc, const struct bt_has_preset_record *record);
+
+	/**
+	 * @brief Callback function for preset deletion.
+	 *
+	 * The callback provides a range of presets that have been deleted by the remote server.
+	 *
+	 * @param conn Bluetooth connection object.
+	 * @param start_index Preset start index that has been deleted.
+	 * @param end_index Preset end index that has been deleted.
+	 */
+	void (*remove)(struct bt_hap_harc *harc, uint8_t start_index, uint8_t end_index);
```

Comment:
`that has been deleted.`
But the callback is called `remove`. 

It is unclear to me whether this is supposed to tell the application to remove something, or tell the application that something has been remove**d** (or deleted).

Reply:
It tells the application that something has been removed on the HAS Server, thus the app is obliged to remove the cached entry (if stored).

The callbacks are named like this because I wanted them to look like operations on some preset cache list.
So if something has been changed on remote, the stack instructs the app to apply the changes on local cache.

Reply:
Right. I'd perhaps modify the text to reflect that more explicitly:

"index that has been deleted on the remote server." so it doesn't indicate that something locally has already been removed.